File: A2/command-node-service/src/events/eventSchema.js
// Event Schema Definitions

const SENSOR_EVENTS = {
  CREATED: 'sensor.created',
  UPDATED: 'sensor.updated',
  ALERT: 'sensor.alert',
  TELEMETRY: 'sensor.telemetry'
};

const COMMAND_EVENTS = {
  ISSUED: 'command.issued',
  MISSION_STATUS_CHANGED: 'mission.status.changed',
  THREAT_LEVEL_CHANGED: 'threat.level.changed',
  ALERT_ACKNOWLEDGED: 'alert.acknowledged'
};

function createSensorEvent(eventType, sensorData, source = 'sensor-service') {
  return {
    event_type: eventType,
    event_id: `event_${Date.now()}_${Math.random().toString(36).substr(2, 9)}`,
    timestamp: new Date().toISOString(),
    source: source,
    data: sensorData
  };
}

function createCommandEvent(eventType, commandData, source = 'command-service') {
  return {
    event_type: eventType,
    event_id: `event_${Date.now()}_${Math.random().toString(36).substr(2, 9)}`,
    timestamp: new Date().toISOString(),
    source: source,
    data: commandData
  };
}

module.exports = {
  SENSOR_EVENTS,
  COMMAND_EVENTS,
  createSensorEvent,
  createCommandEvent
};

File: A4/services/python-command-service/src/events/event_schema.py
import json
import time
import random
import string
from datetime import datetime
from typing import Dict, Any

# Event type constants
SENSOR_EVENTS = {
    'CREATED': 'sensor.created',
    'UPDATED': 'sensor.updated',
    'ALERT': 'sensor.alert',
    'TELEMETRY': 'sensor.telemetry'
}

COMMAND_EVENTS = {
    'ISSUED': 'command.issued',
    'MISSION_STATUS_CHANGED': 'mission.status.changed',
    'THREAT_LEVEL_CHANGED': 'threat.level.changed',
    'ALERT_ACKNOWLEDGED': 'alert.acknowledged'
}


def create_sensor_event(event_type: str, sensor_data: Dict[str, Any], source: str = 'sensor-service') -> Dict[str, Any]:
    return {
        'event_type': event_type,
        'event_id': f"event_{int(time.time() * 1000)}_{''.join(random.choices(string.ascii_lowercase + string.digits, k=9))}",
        'timestamp': datetime.utcnow().isoformat() + 'Z',
        'source': source,
        'data': sensor_data
    }


def create_command_event(event_type: str, command_data: Dict[str, Any], source: str = 'command-service') -> Dict[str, Any]:
    return {
        'event_type': event_type,
        'event_id': f"event_{int(time.time() * 1000)}_{''.join(random.choices(string.ascii_lowercase + string.digits, k=9))}",
        'timestamp': datetime.utcnow().isoformat() + 'Z',
        'source': source,
        'data': command_data
    }
